java/code/src/com/redhat/rhn/manager/org/MigrationManager.java

```diff
@@ -191,6 +193,23 @@ public static void updateAdminRelationships(Org fromOrg, Org toOrg, Server serve
         }
     }
 
+    /**
+     * Updates Entitlements for system migration
+     *
+     * @param server the server to migrate
+     * @param toOrg the organization to migrate to
+     */
+    public static void updateServerEntitlements(Org toOrg, Server server) {
+        server.getEntitlements().forEach(ent -> {
+            // We need to get the current server group id for the given entitlement
+            // to account for the possibility of incorrectly assigned server groups
+            // due to missing entitlement updates in the past.
+            Long fromId = ServerFactory.lookupServerGroupIdByEntitlement(server, ent);
+            Long toId = ServerGroupFactory.lookupEntitled(ent, toOrg).getId();
+            ServerFactory.updateEntitlementsForMigration(server, fromId, toId);
+        });
```

Comment:
Wouldn't something like this work? Then we wouldn't need to add almost anything. We also have `SystemManager.addServersToGroup/removeServersFromGroup`, but I didn't use them on purpose - they do more things and I'm not sure if we want to execute them too (would be worth checking).

```suggestion
        server.getEntitledGroups().forEach(oldGroup -> {
            ServerFactory.removeServerFromGroup(server, oldGroup);
            EntitlementServerGroup newGroup = ServerGroupFactory.lookupEntitled(oldGroup.getGroupType().getAssociatedEntitlement(), toOrg);
            ServerFactory.addServerToGroup(server, newGroup);
        });
```

Reply:
Thanks for the feedback. I updated the PR accordingly please have another look :)